### DISCUSSION TITLE
Can the VPS internally implement a keep-alive feature similar to Vue and React Router

### DISCUSSION THREAD

**Discussion by algesthesiah** [vikejs/vike#756] [Discussion]
Can the VPS internally implement a keep-alive feature similar to Vue and React Router? That is, switching the routing URL under the current browser tab will retain the original page instead of re-rendering it. Design to generate routes using existing file directories simultaneously.

**Comment by brillout** [comment]
I'm not sure I understand what you want to achieve, but have a look at https://vite-plugin-ssr.com/client-routing.

**Reply by algesthesiah** [reply]
Currently using client-routing, I have looked through the client-routing documentation but have not found any example documents similar to implementing https://vuejs.org/guide/built-ins/keep-alive.html.

**Reply by algesthesiah** [reply]
My understanding is that this needs to be actively implemented by the API, as currently each client route will be re-rendered.

**Reply by brillout** [reply]
Are you using React? In that case I believe the idiomatic way to achieve that is to use a global state. Also see the upcoming nested layout feature https://github.com/brillout/vite-plugin-ssr/issues/346.

**Reply by algesthesiah** [reply]
You're so handsome, bro. It's the effect I was looking for.
